cmake/match.py

```diff
@@ -5,77 +5,131 @@
 # See LICENSE.TXT
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-# check if all lines match in a file
-# lines in a match file can contain regex inside of double curly braces {{}}
+# Check if all input file content matches match file content.
+# Lines in a match file can contain regex inside of double curly braces {{}}.
+# Regex patterns are limited to single line.
+#
+# List of available special tags:
+# {{OPT}}    - makes content in the same line as the tag optional
+# {{IGNORE}} - ignores all content until next match or end of input file
```

Comment:
just confirming - by "next match" you mean "next IGNORE tag"? perhaps you could be more specific, please

Reply:
By next match I mean the next line that's successfully matched, next `{{IGNORE}}` tag will do nothing since `{{IGNORE}}` tag is already in effect.

Reply:
och, I see, then please write that in the comment 😉 `the next successfully matched line`

btw, I'd write: `or the end of the input ...` (missing "the" x2)